http/http.go

```diff
@@ -174,6 +180,19 @@ func newRequestWithHeaderSecret(url string, method string, key string, value str
 	return request, nil
 }
 
+func newRequestWithHeaders(url string, method string, headers map[string]string) (*http.Request, error) {
```

Comment:
Please use the new newRequestWithHeaders function inside the existing newRequestWithHeaderSecret function to avoid duplicate code.

Reply:
Did so in the next commit. I even removed the function, because it added almost no abstraction.

Agree?